toolbox/swiftsim/generate_splitting_information.py

```diff
@@ -0,0 +1,416 @@
+        A dictionary holding information used to determine where the 
+        particle data is located, what are the output numbers to analyse,
+        and where to save the information.
+    '''
+    config = {}
+
+    with open(config_path) as file:
+        for line in file:
+            if 'MinSnapshotIndex' in line:
+                config['MinSnapshotIndex'] = int(line.split()[-1])
+            if 'MaxSnapshotIndex' in line:
+                config['MaxSnapshotIndex'] = int(line.split()[-1])
+            if 'SnapshotIdList' in line:
+                config['SnapshotIdList'] = np.array(line.split()[1:]).astype(int)
+            if 'SnapshotPath' in line:
+                config['SnapshotPath'] = line.split()[-1]
+            if 'SnapshotFileBase' in line:
+                config['SnapshotFileBase'] = line.split()[-1]
+            if 'SnapshotDirBase' in line:
+                config['SnapshotDirBase'] = line.split()[-1]
+            if 'SubhaloPath' in line:
+                config['SubhaloPath'] = line.split()[-1]
+
+    return config
+
+def generate_path_to_snapshot(config, snapshot_index):
+    '''
+    Returns the path to the virtual file of a snapshot to analyse.
+
+    Parameters
+    ----------
+    config : dict
+        Dictionary holding relevant infomation about the SWIFT run;
+        loaded previously by load_hbt_config.
+    snapshot_index : int
+        Number of the snapshot to analyse, given relative to the subset
+        analysed by HBT.
+
+    Returns
+    -------
+    str
+        Path to the virtual file of the snapshot.
+    '''
+    if 'SnapshotDirBase' in config: 
+        subdirectory = f"{config['SnapshotDirBase']}_{config['SnapshotIdList'][snapshot_index]:04d}"
+    else:
+        subdirectory = "" 
+
+    return f"{config['SnapshotPath']}/{subdirectory}/{config['SnapshotFileBase']}_{config['SnapshotIdList'][snapshot_index]:04d}.hdf5"
+
+def load_snapshot(file_path):
+    '''
+    Returns the information required to reconstruct which particles were split and which ones
+    are its descendants. Does not provide information about particles that have never split.
+
+    Parameters
+    ----------
+    file_path : str
+        Location of the snapshot to load.
+
+    Returns
+    -------
+    split_counts : np.ndarray
+        Number of splits that have occured along the splitting tree of a given particle.
+    split_trees : np.ndarray
+        Binary tree representing whether a particle changed its ID (1) or not (0)
+        during a split event. The value is represented in base ten.
+    split_progenitors : np.ndarray
+        ID of the particle originally present in the simulation that is the progenitor
+        of the current particle. Used to group all particles that share this common 
+        particle progentor into distinct split trees
+    split_particle_ids : np.ndarray
+        ID of the particle.
+    '''
+
+    # Load snapshot. TODO: consider using MPI.
+    snap = sw.load(file_path)
+
```

Comment:
I'd prefer not to use swiftsimio if possible, just to have less dependencies. In that case you would need to check to make sure you're not being passed a chunk file.

Reply:
The MPI version I intend on writing for #48 will not rely on swiftsimio.